### PR TITLE
Fixes #672 - Set output format at config.yaml Level 

### LINKED ISSUE #672: Set output format at config.yaml Level
Coming from using Sceptre v1 a lot to 2.0 I have noticed the move towards favouring text output as the default output_format rather than JSON as it was before. 

It took me a little while to figure out that a lot of my commands looked broken as I was not setting an `--ouput` flag on my commands to JSON (or YAML) and since getting used to it, It has occurred to me that at a project level and even at a Stack level, it would be really useful to be able to set this once and not have to worry about setting a flag each time.

I would like to be able to set this using something like `output` or `output_format` at my `config.yaml` files and then utilise this as my preferred option over setting --output.

Cascading config should still work and maybe if --ouput is set, it overrides any value in the config.

### PR DESCRIPTION
**Pull request by nmyster**
Coming from using Sceptre v1 a lot to 2.0 I have noticed the move towards favouring text output as the default output_format rather than JSON as it was before.

It took me a little while to figure out that a lot of my commands looked broken as I was not setting an --ouput flag on my commands to JSON (or YAML) and since getting used to it, It has occurred to me that at a project level and even at a Stack level, it would be really useful to be able to set this once and not have to worry about setting a flag each time.

I would like to be able to set this using something like output or output_format at my `config.yam` files and then utilise this as my preferred option over setting `--output`.

### Example
```yaml
region: eu-west-1
project_code: some-project
template_bucket: some-bucket
template_prefix: some-prefix
output: yaml 
or
output: json
```

Cascading config should still work and maybe if `--ouput` is set, it overrides any value in the config.
## PR Checklist

- [x] Wrote a good commit message & description [see guide below].
- [ ] Commit message starts with `[Resolve #issue-number]`.
- [ ] Added/Updated unit tests.
- [ ] Added/Updated integration tests (if applicable).
- [ ] All unit tests (`make test`) are passing.
- [x] Used the same coding conventions as the rest of the project.
- [ ] The new code passes flake8 (`make lint`) checks.
- [x] The PR relates to _only_ one subject with a clear title.
      and description in grammatically correct, complete sentences.

## Approver/Reviewer Checklist

- [ ] Before merge squash related commits.

## Other Information

[Guide to writing a good commit](http://chris.beams.io/posts/git-commit/)

### CODE REVIEW

**Comment by nmyster** [comment]
Looking for some help to add appropriate testing

**Comment by nmyster** [comment]
Commit currently utilises similar code to `s3_details` to get the output value and overrides the context.output_format if it is provided

**Comment by nmyster** [comment]
@ngfgrant  looking for some input on this PR and some assistance getting appropriate testing in place. 
tl;dr - I'd like to set the ouput (json or yaml) in the config.yaml - this PR allows that but it may be a bit hacky.

Have some ideas in general to make the config.yaml more capable so future values may be good to parse - is this a good time to think about how to refactor parts of the code to make this easier in the future

Example future values I'd like to add: profile mappings so that when a profile is provided to different parts of the stacks, it can read the profile mapping to understand what role to assume when run in a CICD environment (where profiles don't exist in the same fashion as running locally)

**Comment by ngfgrant** [comment]
Thanks @nmyster will try get a look soon. Main focus for me is to get 2.1.1 out.

**Comment by ngfgrant** [comment]
Hey @nmyster 

Based on your feedback here, from #709 and in v2.1.4 onward the default output on the CLI will be JSON.

Had a look at this and think it could probably be achieved in a more straightforward way. If we still wanted to set this in the config, which I am not 100% sure we want to do. I think probably this better belongs in a project-wide config file e.g. `sceptre.yaml`. Basically anything that is in `config.yaml` should probably relate to the values of the stack being deployed to CloudFormation.

If we close this one out we could look at project wide config which is definitely something that would be worth doing?

**Comment by ngfgrant** [comment]
Knew we had something regarding project-wide settings https://github.com/cloudreach/sceptre/issues/152

**Comment by ngfgrant** [comment]
Going to close this out in favour of #152 but will take this functionality into any future solution. Thanks again @nmyster
